dss/util/retry.py

```diff
@@ -202,29 +206,31 @@ def wrapper(*args, **kwargs):
                 try:
                     return f(*args, **kwargs)
                 except BaseException as e:
-                    tx.logger.warning("An exception occurred in '%s' of %r", name, tx, exc_info=True)
+                    tx.log_exception(e)
                     if self.retryable(e):
                         attempt = next(attempts, None)
                         if attempt is None:
-                            tx.logger.debug("Exceeded retry limit for '%s' of %r", name, tx)
+                            tx.logger.info("Exceeded retry limit for %r.", tx)
```

Comment:
If a retry limit is exceeded, I am inclined to think it should be logged at `warning` rather than `info`, should it not?

Reply:
I was going back and forth on this but I think you are right.